src/Mapper/ArrayToObjectMapper.php

```diff
@@ -245,27 +246,20 @@ private function validate(object|string $object): void
         $validator->validate($object);
     }
 
-    /**
-     * @param class-string<DateTime|DateTimeImmutable|DateTimeInterface> $type
-     * @param ReflectionProperty $property
-     *
-     * @return Caster
-     */
-    private function createDateCaster(string $type, ReflectionProperty $property): Caster
+    private function validateInferredRules(ReflectionProperty $property, mixed $value): void
     {
-        $format = null;
-
-        try {
-            $format = attribute(DateTimeFormat::class)
-                ->in($property)
-                ->first();
-        } catch (Throwable) {
+        $failingRules = [];
 
+        foreach (get(InferrerConfig::class)->inferrers as $inferrer) {
+            foreach ($inferrer->infer($property, $value) as $rule) {
```

Comment:
Can you explain the goal of inferrers?

Reply:
Sure thing!

The goal of inferrers is to provide a convenient way to register automatic validation rules for types. It's inspired by my fav feature of laravel-data =)

For example - in this PR we have ModelId, which is a simple vo, and in our system it has to start with 5 for some reason - we have registered the inferrer, now any time a new ModelId is created from a request, we are going to run that validation automatically.

This is especially useful in big apps where you use value objects. It saves tons of time and encourages developers to create stricter types, cuz you get the validation *everywhere* for free - people don't even have to know that it happens which is great if you add new people to the project and you want to ensure safety.

If you don't like the callback style, we could possibly just register inferrers as a key => value in the config, so it's slightly faster to infer the rules, we'd lose some flexibility though.

Reply:
On top of that - before this change, if we sent a JSON request to an endpoint:

```
{
abc: [[[]]]
}
```

that has request class with

public bool $abc;

It'd just be casted to bool without any problems - which I think is safe to assume is not what we'd want to happen.